rust/azure_iot_operations_connector/src/file_mount_azure_device_registry.rs

```diff
@@ -1,4 +1,1110 @@
+        let mut debouncer = new_debouncer(
+            debounce_duration,
+            None,
+            move |res: Result<Vec<notify_debouncer_full::DebouncedEvent>, Vec<notify::Error>>| {
+                match res {
+                    Ok(events) => {
+                        // TODO: There might be a case where we receive events out of order (i.e a modify before a create). The file mount map accounts for this but
+                        // it might be better to just match on an Any.
+                        for debounced_event in &events {
+                            match debounced_event.event.kind {
+                                EventKind::Create(event::CreateKind::File) => { // Event signals the creation of a device endpoint
+                                    debounced_event.paths.iter().for_each(|path| {
+                                        // TODO: We could use an expect here since we are sure the path is valid
+                                        let Some(mount_path) = path.parent() else {
+                                            log::warn!("Failed to get parent path from device endpoint");
+                                            return;
+                                        };
+
+                                        let device = match DeviceEndpointRef::try_from(path) {
+                                            Ok(device) => device,
+                                            Err(err) => {
+                                                log::warn!("Failed to parse device endpoint from path: {err:?}");
+                                                return;
+                                            }
+                                        };
+
+                                        // Insert the device endpoint into the file mount map
+                                        file_mount_map.insert_device_endpoint(&device);
+
+                                        // The create event is for a new file, so we need to get the assets
+                                        let assets =
+                                            match get_asset_names(mount_path, &device) {
+                                                Ok(assets) => assets,
+                                                Err(err) => {
+                                                    log::warn!("Failed to get asset names: {err:?}");
+                                                    return;
+                                                }
+                                            };
+
+                                        // Update the file mount map with the new assets
+                                        file_mount_map.update_assets(&device, &assets);
+                                    });
+                                }
+                                EventKind::Modify(_) => { // Event signals the creation or deletion of an asset
```

Comment:
do we care about the contents of this event?

Reply:
We only care about the paths it gives us